include/cuco/detail/static_multimap.inl

```diff
@@ -0,0 +1,1814 @@
+  CG const& g,
+  value_type const& pair,
+  uint32_t* cg_counter,
+  value_type* probe_output_buffer,
+  value_type* contained_output_buffer,
+  atomicT* num_matches,
+  OutputZipIt1 probe_output_begin,
+  OutputZipIt2 contained_output_begin,
+  PairEqual pair_equal) noexcept
+{
+  constexpr bool is_outer = false;
+  pair_retrieve_impl<cg_size, buffer_size, is_outer>(g,
+                                                     pair,
+                                                     cg_counter,
+                                                     probe_output_buffer,
+                                                     contained_output_buffer,
+                                                     num_matches,
+                                                     probe_output_begin,
+                                                     contained_output_begin,
+                                                     pair_equal);
+}
+
+template <typename Key,
+          typename Value,
+          cuda::thread_scope Scope,
+          class ProbeSequence,
+          typename Allocator>
+template <uint32_t cg_size,
+          uint32_t buffer_size,
+          typename CG,
+          typename atomicT,
+          typename OutputZipIt1,
+          typename OutputZipIt2,
+          typename PairEqual>
+__device__ void
+static_multimap<Key, Value, Scope, ProbeSequence, Allocator>::device_view::pair_retrieve_outer(
+  CG const& g,
+  value_type const& pair,
+  uint32_t* cg_counter,
+  value_type* probe_output_buffer,
+  value_type* contained_output_buffer,
+  atomicT* num_matches,
+  OutputZipIt1 probe_output_begin,
+  OutputZipIt2 contained_output_begin,
+  PairEqual pair_equal) noexcept
+{
+  constexpr bool is_outer = true;
+  pair_retrieve_impl<cg_size, buffer_size, is_outer>(g,
+                                                     pair,
+                                                     cg_counter,
+                                                     probe_output_buffer,
+                                                     contained_output_buffer,
+                                                     num_matches,
+                                                     probe_output_begin,
+                                                     contained_output_begin,
+                                                     pair_equal);
+}
+
+template <typename Key,
+          typename Value,
+          cuda::thread_scope Scope,
+          class ProbeSequence,
+          typename Allocator>
+std::size_t static_multimap<Key, Value, Scope, ProbeSequence, Allocator>::get_size(
+  cudaStream_t stream) const noexcept
+{
+  auto begin = thrust::make_transform_iterator(
+    raw_slots(), [] __device__(cuco::pair_type<Key, Value> const& pair) { return pair.first; });
+  slot_is_filled<Key> filled(empty_key_sentinel_);
+
+  return thrust::count_if(thrust::cuda::par.on(stream), begin, begin + capacity_, filled);
```

Comment:
I think we should add header `<thrust/count.h>` for `count_if`, and header `<thrust/iterator/transform_iterator.h>` for `make_transform_iterator`.